metastats-domain/src/main/groovy/life/qbic/metastats/exceptions/InvalidFileNameException.groovy

```diff
@@ -0,0 +1,26 @@
+package life.qbic.metastats.exceptions
+
+/**
+ * Invalid file names throw an exception
+ *
+ * If a filename does not follow the naming conventions this exception should be thrown
+ *
+ * @since: 1.0
+ * @author: Jennifer Bödker
+ *
+ */
+class InvalidFileNameException extends IllegalArgumentException{
```

Comment:
Instead of making your class dependent on the IllegalArgumentException, you usually would extend as a unchecked `RuntimeException`. If you want to state an illegal argument from a method input, then you can directly throw the exception `IllegalArgumentException`.

Please make also sure to override all super constructors.

Reply:
Ok, I change the extending exception class and added the constructors: 8814549
is this how you meant it?

Reply:
yes, perfect!